packages/samples/headless-ssr-commerce/app/_components/cart.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Cart as CartController,
+  CartItem,
+  CartState,
+  ContextState,
+} from '@coveo/headless/ssr-commerce';
+import {useEffect, useState, FunctionComponent} from 'react';
+import {formatCurrency} from '../_utils/format-currency';
+
+interface CartProps {
+  staticState: CartState;
+  controller?: CartController;
+  staticContextState: ContextState;
+}
+
+export const Cart: FunctionComponent<CartProps> = ({
+  staticState,
+  controller,
+  staticContextState,
+}) => {
+  const [state, setState] = useState(staticState);
+
+  useEffect(
+    () => controller?.subscribe(() => setState({...controller.state})),
```

Comment:
Why is the state being spread?
```suggestion
    () => controller?.subscribe(() => setState(controller.state)),
```

Reply:
Not sure, I honestly just copied this component from fred's react sample ! I think it is to make sure to create another object. React can sometimes not detect the change for objects and not rerender.

Reply:
For state objects its better to spread to make sure the nested properties changed are detected. https://react.dev/reference/react/useState#examples-objects